src/v/compression/include/compression/lz4_decompression_buffers.h

```diff
@@ -100,4 +107,54 @@ class lz4_decompression_buffers {
 std::ostream& operator<<(
   std::ostream&, lz4_decompression_buffers::alloc_ctx::allocation_state);
 
+inline lz4_decompression_buffers::alloc_ctx::allocation_state operator|(
+  lz4_decompression_buffers::alloc_ctx::allocation_state a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    using t = std::underlying_type_t<
+      lz4_decompression_buffers::alloc_ctx::allocation_state>;
+    return lz4_decompression_buffers::alloc_ctx::allocation_state(t(a) | t(b));
+}
+
+inline void operator|=(
+  lz4_decompression_buffers::alloc_ctx::allocation_state& a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    a = (a | b);
+}
+
+inline lz4_decompression_buffers::alloc_ctx::allocation_state operator&(
+  lz4_decompression_buffers::alloc_ctx::allocation_state a,
+  lz4_decompression_buffers::alloc_ctx::allocation_state b) {
+    using t = std::underlying_type_t<
+      lz4_decompression_buffers::alloc_ctx::allocation_state>;
+    return lz4_decompression_buffers::alloc_ctx::allocation_state(t(a) & t(b));
+}
```

Comment:
Instead of writing this out explicitly, why not use `bitset<4>`? you can even assign names, like this:

```
enum class states { a, b, c, d, num_flags };
std::bitset<states::num_flags> flags;
```

and then all these operations can be done with bitset operations?

Reply:
I started exploring the use of a bitset because it looked like a good fit, but then I realized it contains just plain ints, so we lose the type safety of the enum class, and the compiler warnings from incomplete switch statements etc. that enum class provides.

I did move the bitwise operators to an anonymous namespace in the source file so they are not part of the public API.

Reply:
This makes sense. No need to swap out for bitset.